### PR TITLE
allow to provide required info for beta review

### PR DESCRIPTION
**Pull request by SAVeselovskiy**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] I've updated the documentation if necessary.

### Description

New option for pilot action, that allows to provide required contact information for testflight beta review. Now you can call pilot as follows:
```
pilot(changelog: testflight_changelog,
      beta_app_description:appDescription,
      distribute_external: true,
      beta_app_feedback_email: "mail@gmail.com",
      review_contact_info: {
         review_first_name: "MyTestAccount",
         review_last_name: "MyTestAccount",
         review_phone_number: "+12345678901",
         review_contact_email: "mail@gmail.com"
      })
```
### Motivation and Context

Pilot raise error when trying to submit new build for testflight beta app review without required contact information. Issue #7085 

https://github.com/fastlane/fastlane/issues/7085

I tested my changes on my application by executing fastlane action. Now iTC doesn't raise error. When i run `bundle exec rspec` on fastlane right after git clone it raises error ```Scan Scan::DetectValues Xcode config handling fetches the path from the Xcode config
     Failure/Error: expect(derived_data).to match(%r{/.*Xcode/DerivedData/app-\w*$})```. But there is no changes yet. After editing files i get the same error. I think it's not about my changes.

### CODE REVIEW

**Comment by SAVeselovskiy** [comment]
I signed it!

**Comment by SAVeselovskiy** [comment]
Can't find out how to fix this CircleCI error

**Comment by pav1790** [comment]
These changes are incomplete for someone using Spaceship. Please add the following three build_info : reviewUserName, reviewPassword, reviewAccountRequired.  
[  reviewAccountRequired = true when length(reviewUserName > 0 ) && length(reviewPassword) > 0  ]

**Comment by ohayon** [comment]
@SAVeselovskiy thanks for the PR! Would you mind adding some tests? 👍 🐙

File: pilot/lib/pilot/options.rb [hunk 1]
```diff
@@ -52,6 +52,12 @@ def self.available_options
                                      optional: true,
                                      env_name: "PILOT_BETA_APP_FEEDBACK",
                                      description: "Provide the beta app email when uploading a new build"),
+        FastlaneCore::ConfigItem.new(key: :review_contact_info,
+                                     short_option: "-y",
+                                     optional: true,
+                                     env_name: "PILOT_BETA_APP_CONTACT_REVIEW_LAST_NAME",
```
Comment (KrauseFx) [inline pilot/lib/pilot/options.rb]: This shouldn't be called `LAST_NAME`

File: pilot/lib/pilot/options.rb [hunk 2]
```diff
@@ -52,6 +52,12 @@ def self.available_options
...
+                                     is_string: false,
+                                     description: "Provide the beta app contact information review_last_name when uploading a new build to beta review"),
```
Comment (KrauseFx) [inline pilot/lib/pilot/options.rb]: This line would need some more information on how and what data to provide 👍

File: spaceship/lib/spaceship/tunes/tunes_client.rb [hunk 3]
```diff
@@ -1163,18 +1168,18 @@ def create_tester!(tester: nil, email: nil, first_name: nil, last_name: nil, gro
...
-        lastName: {
-          value: last_name || ""
-        },
-        testing: {
-          value: true
-        }
+          emailAddress: {
+              value: email
+          },
+          firstName: {
```
Comment (KrauseFx) [inline spaceship/lib/spaceship/tunes/tunes_client.rb]: Why do we have so many white space changes here?

**Review by KrauseFx — CHANGES_REQUESTED** [review]
Hey @SAVeselovskiy, thanks for taking the time to submit this PR, that's an extremely useful feature for _pilot_ as part of _fastlane_, great job! 

Have you tested this locally on your machine for some of your apps? As @ohayon mentioned, we'd need at least one test case covering this new feature, so that we ensure it keeps working in the future.

Thanks for all this ❤️

**Comment by KrauseFx** [comment]
Mh, the failure is weird, can you try pushing another commit and see if that works? Seems like Circle can't find your commit

File: pilot/lib/pilot/options.rb [hunk 4]
```diff
@@ -52,6 +52,12 @@ def self.available_options
                                      optional: true,
                                      env_name: "PILOT_BETA_APP_FEEDBACK",
                                      description: "Provide the beta app email when uploading a new build"),
+        FastlaneCore::ConfigItem.new(key: :review_contact_info,
+                                     short_option: "-y",
```
Comment (giginet) [inline pilot/lib/pilot/options.rb]: What do it mean `-y`?

File: pilot/lib/pilot/options.rb [hunk 5]
```diff
@@ -52,6 +52,12 @@ def self.available_options
                                      optional: true,
                                      env_name: "PILOT_BETA_APP_FEEDBACK",
                                      description: "Provide the beta app email when uploading a new build"),
+        FastlaneCore::ConfigItem.new(key: :review_contact_info,
```
Comment (giginet) [inline pilot/lib/pilot/options.rb]: you should add `type` parameter.

`type: Hash,`

**Comment by KrauseFx** [comment]
It seems like this needs to be re-written to work with the latest iTunes Connect, right? 😢

**Comment by KrauseFx** [comment]
Yeah, unfortunately, we have to close this PR, as it conflicts with our other recent changes to TestFlight.

**Comment by KrauseFx** [comment]
Thanks for your contribution @SAVeselovskiy, feel free to fork the current `master` branch and apply the changes there, if necessary

**Pull request closed without merging**